x-pack/solutions/chat/plugins/workchat-app/server/services/orchestration/graph_events/build_events_converter.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { StreamEvent as LangchainEvent } from '@langchain/core/tracers/log_stream';
+import { AgentRunEvents } from '../types';
+import {
+  EventConverter,
+  getSimpleGraphConverter,
+  getSearchAgentGraphConverter,
+} from './converters';
+import { graphNames } from '../constants';
+
+export type GraphEventConverter = (event: LangchainEvent) => AgentRunEvents[];
+
+/**
+ * Build the event converter based on the agent graph definition.
+ *
+ * Note: this is totally static for now, but will become dynamic once we introduce assistant
+ * configuration / workflow builder functionalities.
+ */
+export const buildEventsConverter = (): GraphEventConverter => {
+  const converters: EventConverter[] = [];
+
+  // main graph converter
+  converters.push(getSimpleGraphConverter({ graphName: graphNames.mainAgent }));
+
+  // search agent converter
+  converters.push(getSearchAgentGraphConverter({ graphName: graphNames.searchAgent }));
+
+  return function graphEventConverter(event: LangchainEvent): AgentRunEvents[] {
+    for (const converter of converters) {
+      if (converter.handleEvent(event)) {
+        return converter.convert(event);
+      }
+    }
+    return [];
+  };
+};
```

Comment:
Hardcoded for now. Later we know we will want each graph node to have an associated event converter